braindecode/datasets/hub.py

```diff
@@ -0,0 +1,1059 @@
+                    raise ValueError(
+                        f"Inconsistent channel names: dataset 0 has {first_ch_names} "
+                        f"but dataset {i} has {ds.windows.ch_names}"
+                    )
+                if ds.windows.info["sfreq"] != first_sfreq:
+                    raise ValueError(
+                        f"Inconsistent sampling frequencies: dataset 0 has {first_sfreq} Hz "
+                        f"but dataset {i} has {ds.windows.info['sfreq']} Hz. "
+                        f"Please resample all datasets to a common frequency before saving. "
+                        f"Example:\n"
+                        f"  from braindecode.preprocessing import preprocess, Preprocessor\n"
+                        f"  preprocessors = [Preprocessor('resample', sfreq={first_sfreq})]\n"
+                        f"  preprocess(concat_ds, preprocessors)"
+                    )
+            elif dataset_type == "EEGWindowsDataset":
+                if ds.raw.ch_names != first_ch_names:
+                    raise ValueError(
+                        f"Inconsistent channel names: dataset 0 has {first_ch_names} "
+                        f"but dataset {i} has {ds.raw.ch_names}"
+                    )
+                if ds.raw.info["sfreq"] != first_sfreq:
+                    raise ValueError(
+                        f"Inconsistent sampling frequencies: dataset 0 has {first_sfreq} Hz "
+                        f"but dataset {i} has {ds.raw.info['sfreq']} Hz. "
+                        f"Please resample all datasets to a common frequency before saving. "
+                        f"Example:\n"
+                        f"  from braindecode.preprocessing import preprocess, Preprocessor\n"
+                        f"  preprocessors = [Preprocessor('resample', sfreq={first_sfreq})]\n"
+                        f"  preprocess(concat_ds, preprocessors)"
+                    )
+            elif dataset_type == "RawDataset":
+                if ds.raw.ch_names != first_ch_names:
+                    raise ValueError(
+                        f"Inconsistent channel names: dataset 0 has {first_ch_names} "
+                        f"but dataset {i} has {ds.raw.ch_names}"
+                    )
+                if ds.raw.info["sfreq"] != first_sfreq:
+                    raise ValueError(
+                        f"Inconsistent sampling frequencies: dataset 0 has {first_sfreq} Hz "
+                        f"but dataset {i} has {ds.raw.info['sfreq']} Hz. "
+                        f"Please resample all datasets to a common frequency before saving. "
+                        f"Example:\n"
+                        f"  from braindecode.preprocessing import preprocess, Preprocessor\n"
+                        f"  preprocessors = [Preprocessor('resample', sfreq={first_sfreq})]\n"
+                        f"  preprocess(concat_ds, preprocessors)"
+                    )
+
+        # Calculate dataset size
```

Comment:
nice to to calculate the total size! Just a few fixes on the actual calculations below: